doctors-portal/src/API/RSBC.DMF.DoctorsPortal.API/CmsAdapterConfiguration.cs

```diff
@@ -29,29 +29,40 @@ public static IServiceCollection AddCmsAdapterGrpcService(this IServiceCollectio
                 var httpClient = new HttpClient(httpClientHandler);
                 // set default request version to HTTP 2.  Note that Dotnet Core does not currently respect this setting for all requests.
                 httpClient.DefaultRequestVersion = HttpVersion.Version20;
-
-                var initialChannel = GrpcChannel.ForAddress(serviceUrl, new GrpcChannelOptions { HttpClient = httpClient });
-
-                var initialClient = new CaseManager.CaseManagerClient(initialChannel);
-                // call the token service to get a token.
-                var tokenRequest = new TokenRequest { Secret = clientSecret };
-
-                var tokenReply = initialClient.GetToken(tokenRequest);
-
-                if (tokenReply != null && tokenReply.ResultStatus == ResultStatus.Success)
+                if (string.IsNullOrEmpty (clientSecret))
                 {
-                    // Add the bearer token to the client.
-                    httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {tokenReply.Token}");
-
+                    // add the service without authentication.
                     var channel = GrpcChannel.ForAddress(serviceUrl, new GrpcChannelOptions { HttpClient = httpClient });
 
                     services.AddTransient(_ => new CaseManager.CaseManagerClient(channel));
                     services.AddTransient(_ => new UserManager.UserManagerClient(channel));
                 }
                 else
                 {
-                    Log.Logger.Information("Error getting token for Case Management Service");
+                    var initialChannel = GrpcChannel.ForAddress(serviceUrl, new GrpcChannelOptions { HttpClient = httpClient });
+
+                    var initialClient = new CaseManager.CaseManagerClient(initialChannel);
+                    // call the token service to get a token.
+                    var tokenRequest = new TokenRequest { Secret = clientSecret };
+
+                    var tokenReply = initialClient.GetToken(tokenRequest);
+
+                    if (tokenReply != null && tokenReply.ResultStatus == ResultStatus.Success)
+                    {
+                        // Add the bearer token to the client.
+                        httpClient.DefaultRequestHeaders.Add("Authorization", $"Bearer {tokenReply.Token}");
+
+                        var channel = GrpcChannel.ForAddress(serviceUrl, new GrpcChannelOptions { HttpClient = httpClient });
+
+                        services.AddTransient(_ => new CaseManager.CaseManagerClient(channel));
+                        services.AddTransient(_ => new UserManager.UserManagerClient(channel));
+                    }
+                    else
+                    {
+                        Log.Logger.Information("Error getting token for Case Management Service");
```

Comment:
Discussed, and this is a valid point.  I will look into adjusting this behavior so that it is a requirement for PROD startup; there may be a need to startup in DEV without all services